spec/controllers/settings/dex_connector_oidcs_controller_spec.rb

```diff
@@ -0,0 +1,371 @@
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "good oidc",
+            provider_url:  good_provider,
+            callback_url:  "http://well.formed.but.invalid/", # can't be blank in form
+            basic_auth:    true, # also can't be blank - only true/false
+            client_id:     "client",
+            client_secret: ""
+          }
+        end
+        expect(response).to render_template("new")
+        expect(response.body).to match(/Client Secret can('|&#39;)t be blank/i)
+        # TODO: expect save button to be disabled
+      end
+
+      # rubocop:disable RSpec/MultipleExpectations
+      it "shows an error message for non-http issuer entry" do
+        get :create, validate: false, dex_connector_oidc: {
+          name:          "good oidc",
+          provider_url:  "bare.hostname",
+          callback_url:  "http://well.formed.but.invalid/",
+          basic_auth:    true,
+          client_id:     "client",
+          client_secret: "secret"
+        }
+        expect(response).to render_template("new")
+        expect(response.body).to match(/is not a valid HTTP URL/)
+        expect(response.body).not_to match(/unresolvable hostname|discovery failure|timeout/)
+      end
+      # rubocop:enable RSpec/MultipleExpectations
+
+      # rubocop:disable RSpec/MultipleExpectations
+      it "shows an error message for invalid OIDC issuer hostname" do
+        get :create, validate: false, dex_connector_oidc: {
+          name:          "good oidc",
+          provider_url:  "http://this.fqdn.is.invalid", # RFC 6761
+          callback_url:  "http://well.formed.but.invalid/",
+          basic_auth:    true,
+          client_id:     "client",
+          client_secret: "secret"
+        }
+        expect(response).to render_template("new")
+        expect(response.body).to match(/is not a valid OIDC provider/)
+        expect(response.body).to match(/unresolvable hostname|discovery failure/)
+      end
+      # rubocop:enable RSpec/MultipleExpectations
+
+      # rubocop:disable RSpec/MultipleExpectations
+      it "shows an error message for mismatched OIDC issuer" do
+        VCR.use_cassette("oidc/invalid_connector", allow_playback_repeats: true, record: :none) do
+          get :create, validate: false, dex_connector_oidc: {
+            name:          "good oidc",
+            provider_url:  "http://your.fqdn.here:5556/bad",
+            callback_url:  "http://well.formed.but.invalid/",
+            basic_auth:    true,
+            client_id:     "client",
+            client_secret: "secret"
+          }
+        end
+        expect(response).to render_template("new")
+        expect(response.body).to match(/is not a valid OIDC provider/)
+        expect(response.body).to match(/discovery failure/)
+      end
+      # rubocop:enable RSpec/MultipleExpectations
+
+      # it "shows an error message for 404-generating OIDC issuer"
```

Comment:
This appears to be a todo?

Reply:
It's a comment which was implemented elsewhere and then I forgot to remove it. :)

fixed in e337b2b